### ISSUE TITLE
Update aws_route53_resolver_dnssec_config to be usable and configurable

### ISSUE THREAD

**Issue by aidan-mundy** [hashicorp/terraform-provider-aws#20732]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

There are a few problems with the existing DNSSEC Validation config resource. 
* The timeout is slightly too short (in my experience), so the resource will always be tainted if it tries to enable 
* The resource's documentation does not reflect how it behaves (resolved by fixing next point)
* The resource relies on creation/deletion to enable/disable the configuration rather than a boolean field

The timeout on the resource should be increased from 5 to 10 minutes. A new field should be added to support enabling/disabling as configuration rather than via creation/deletion. 

### New or Affected Resource(s)

* aws_route53_resolver_dnssec_config

### Potential Terraform Configuration

Before: (creating the resource enables validation, deleting disables validation)
```hcl
resource "aws_vpc" "this" {
  cidr_block = "10.0.0.0/16"
}

resource "aws_route53_resolver_dnssec_config" "this" {
  resource_id = aws_vpc.this.id
}
```

After: (creating the resource only makes changes if the state does not match the desired state, deleting does nothing)
```hcl
resource "aws_vpc" "this" {
  cidr_block = "10.0.0.0/16"
}

resource "aws_route53_resolver_dnssec_config" "this" {
  resource_id = aws_vpc.this.id
  enable_validation = BOOLEAN
}
```

### References

* #17012 The MR creating the resource

**Comment by aidan-mundy** [comment]
Just ran back into this issue and I wanted to clarify something from the original text: 
At face value, it doesn't seem very different to have a `count` field operated by a boolean to enable/disable the resource, this does not allow for drift detection in the "off" state. This means that Terraform is not managing the resources state when it believes the DNSSEC is disabled.

**Comment by tokiwong** [comment]
@aidan-mundy #21797 has been merged and is rolled into the v3.66 milestone
